Source/scrollrt.cpp

```diff
@@ -1550,18 +1551,19 @@ void DrawAndBlit()
 		hgt = gnScreenHeight;
 	}
 	DrawXPBar(out);
-	scrollrt_draw_cursor_item(out);
+	DrawCursor(out);
 
 	DrawFPS(out);
 
 	unlock_buf(0);
 
 	DrawMain(hgt, ddsdesc, drawhpflag, drawmanaflag, drawsbarflag, drawbtnflag);
 
+	RenderPresent();
+
 	lock_buf(0);
-	scrollrt_draw_cursor_back_buffer(GlobalBackBuffer());
+	ClearCursor(GlobalBackBuffer());
 	unlock_buf(0);
-	RenderPresent();

```

Comment:
Did you test this both on SDL1 and SDL2?

Reply:
Yeah, on SDL2 we clear the cursor from `pal_surface` (which is always a separate surface there) so the order doesn't matter

Reply:
One issue is that if we're double-buffering in direct rendering mode, we're unable to clear the cursor from the back buffer -- but we can look into this later as the direct rendering feature is still experimental

Reply:
Fixed